include/nighthawk/common/request_source_plugin.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+/**
+ * A factory that must be implemented for each RequestSourcePlugin. It instantiates the specific
+ * RequestSourcePlugin class after unpacking the plugin-specific config proto.
+ */
+class RequestSourcePluginConfigFactory : public Envoy::Config::TypedFactory {
```

Comment:
Rename the file name to something like `request_source_plugin_config_factory.h`?